api/python/pyproject.toml

```
@@ -0,0 +1,149 @@
+[build-system]
+requires = ["uv_build>=0.8.15,<0.9.0"]
+build-backend = "uv_build"
+
+[project]
+name = "quilt3"
+version = "7.0.0"
+description = "Quilt: where data comes together"
+readme = "README.rst"
+requires-python = ">=3.9"
+license = "Apache-2.0"
+license-files = ["LICENSE"]
+authors = [
+    {name = "quiltdata", email = "contact@quiltdata.io"}
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+]
+dependencies = [
+    "platformdirs>=2",
+    "boto3>=1.21.7",
+    "jsonlines==1.2.0",
+    "PyYAML>=5.1",
+    "requests>=2.12.4",
+    "tenacity>=5.1.1,!=8.4.0",
+    "tqdm>=4.32",
+    "requests_futures==1.0.0",
+    "jsonschema>=3,<5",
+    "pydantic>=2.0.0,<3.0.0",
+]
+
+[project.optional-dependencies]
+pyarrow = [
+    "numpy>=1.14.0",
+    "pandas>=0.19.2",
+    "pyarrow>=0.14.1",
+]
+anndata = [
+    "anndata>=0.8.0",
+]
+catalog = [
+    "quilt3_local>=2,<3",
+    "uvicorn>=0.15,<0.18",
+    "aiobotocore[boto3]>=2",
+]
+
+[project.urls]
+Homepage = "https://github.com/quiltdata/quilt"
+Repository = "https://github.com/quiltdata/quilt"
+
+[project.scripts]
+quilt3 = "quilt3.main:main"
+
+[tool.uv.build-backend]
+module-root = ""
+module-name = "quilt3"
+
+[dependency-groups]
+dev = [
+    "pytest==6.*",
+    "pytest-cov",
+    "coverage==6.4",
+    "pytest-env",
+    "pytest-subtests",
+    "responses",
+    "anndata!=0.9.0",
+    "numpy>=1.14.0",
+    "pandas>=0.19.2",
+    "pyarrow>=0.14.1",
```

I think it could be replaced with

```suggestion
    "quilt3[pyarrow,anndata]",
```

these probably should moved to [constraint-dependencies](https://docs.astral.sh/uv/reference/settings/#constraint-dependencies):
```
"coverage==6.4",
"anndata!=0.9.0",
```

or maybe just omit them since they were added as temporary solution and probably not needed anymore